test/tests/functional/pbs_ralter.py

```diff
@@ -0,0 +1,1007 @@
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under a
+# commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+import time
+from tests.functional import *
+
+
+class TestPbsResvAlter(TestFunctional):
+
+    """
+    This test suite is for testing the reservation times modification feature.
+    """
+
+    # Class variables
+    tzone = None
+    # These must be equal to the declarations in pbs_error.h
+    PBSE_UNKRESC = 236
+    PBSE_RESV_NOT_EMPTY = 113
+    PBSE_INCORRECT_USAGE = 2
+    fmt = "%a %b %d %H:%M:%S %Y"
+
+    def setUp(self):
+
+        TestFunctional.setUp(self)
+        # Set PBS_TZID, needed for standing reservation.
+        if 'PBS_TZID' in self.conf:
+            self.tzone = self.conf['PBS_TZID']
+        elif 'PBS_TZID' in os.environ:
+            self.tzone = os.environ['PBS_TZID']
+        else:
+            self.logger.info('Timezone not set, using Asia/Kolkata')
+            self.tzone = 'Asia/Kolkata'
+
+        a = {'resources_available.ncpus': 4}
+        self.server.create_vnodes('vnode', a, num=2, mom=self.mom)
+
+        rc = self.server.manager(MGR_CMD_SET, SERVER, {
+            'log_events': 4095}, expect=True)
+        self.assertTrue(rc)
+
+    def submit_and_confirm_reservation(self, offset, duration, standing=False,
+                                       select="1:ncpus=4",
+                                       rrule="FREQ=HOURLY;COUNT=2",
+                                       ExpectSuccess=1):
+        """
+        Helper function to submit a reservation and wait until it is confirmed.
+        It also checks for the corresponding server and accounting logs.
+
+        param offset: time in seconds after which the reservation should start.
+        type  offset: int
+
+        param duration: duration of the reservation to be submitted.
+        type  duration: int
+
+        param standing: whether to submit a standing reservation. Default: No.
+        type  standing: bool
+
+        param select: select specification for the reservation,
+                      Default: "1:ncpus=4"
+        type  select: string.
+
+        param rrule:  iCal recurrence rule for submitting a standing
+                      reservation.  Default: "FREQ=HOURLY;COUNT=2"
+        type  select: string.
+
+        param ExpectSuccess: Whether the caller expects the reservation
+                             to be successful or not.
+        type  ExpectSuccess: int.
```

Comment:
it is not used as a boolean. I have changed the variable name and provided a description of this argument.